src/wgquickprocess.cpp

```diff
@@ -35,18 +38,18 @@ QString scriptPath() {
 
 }  // namespace
 
-// static
-bool WgQuickProcess::run(
-    Daemon::Op op, const QString& privateKey, const QString& deviceIpv4Address,
+bool WgQuickProcess::validateWgArgs(
```

Comment:
I love this method! Because we can extend it (not now, but in a separate PR)!
For instance, we can replace `VALIDATE` with a proper validation such as:

- keys must have a precise length
- ipv4 addresses must have a precise format
- ipv6 addresses too
- etc

Reply:
Yes! I started adding tests, but then decided not to get ahead of myself as we may not need a validation method like this when we are calling different methods / classes to handle different wg pieces.

Reply:
As I was doing other work, I noticed the parseConfig method in the daemon here: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/daemon/daemon.cpp#L90. It seems to me like we don't need this additional validation given that config is validated there and we pass in everything from config. Any thoughts?